flink-runtime/src/main/java/org/apache/flink/runtime/taskmanager/Task.java

```diff
@@ -879,6 +875,24 @@ else if (transitionState(current, ExecutionState.FAILED, t)) {
 		}
 	}
 
+	private void releaseNetworkResources() throws Exception {
+		LOG.debug("Release task {} network resources (state: {}).",
+			taskInfo.getTaskNameWithSubtasks(), getExecutionState());
+
+		for (ResultPartition partition : producedPartitions) {
+			taskEventDispatcher.unregisterPartition(partition.getPartitionId());
+			partition.close();
```

Comment:
but I think it would be ok, if we actually add one more method `closeNetworkResources` and put there partition/gate closings from `TaskCanceler.run` and use here in `releaseNetworkResources` after `taskEventDispatcher.unregisterPartition` and `if () partition.fail` loops. We will eliminate code duplication and improve log/exception handling in former `unregisterTask`.

Reply:
If you also think so, we could integrate the network release with the process in task canceler. :)